### PR TITLE
Implemented pause/resume on SIGUSR2

### PR DESCRIPTION
**Pull request by antbbn**
This addresses Issue #4 . 
Also commented out there is an implementation that notifies on pause and resume.

I also changed your "infinal loop" because it didn't compile using clang on osx.

### CODE REVIEW

File: spt.c [hunk 1]
```diff
@@ -80,16 +80,28 @@ void
 remaining_time(int sigint)
 {
 	char buf[17];
-	if (signal(SIGUSR1, SIG_IGN) != SIG_IGN)
-		signal(SIGUSR1, remaining_time);
+	//if (signal(SIGUSR1, SIG_IGN) != SIG_IGN)
+	//	signal(SIGUSR1, remaining_time);
```
Comment (antbbn) [inline spt.c:84]: I commented this out because I wasn't sure about its utility.
Is it necessary to reset the handler for SIGUSR1 here?

Comment (pickfire) [inline spt.c:84]: If it's commented out, IIRC it only handle the signal `SIGUSR1` once which means it only display the remaining time once. And I think the spaces below aren't necessary.

Comment (antbbn) [inline spt.c:84]: In my test it keeps handling the signal. Maybe if it doesn't work on your side we can restore it.

**Comment by pickfire** [comment]
@antbbn Thanks a lot!

This is merged in commit `938cd77` but then I think that it should **wait for the signal** instead of just running `timecount += 0`. I will add this to the TODO in the future.

Good luck and have a nice day!

File: spt.c (hunk 1 above)
Comment (pips-) [reply spt.c:84]: Sorry for commenting on a closed PR.

As of manual for signal(2) say, it behavior varies across UNIX versions. I think the need to set the handler again in the handler itself is one of these variant behaviors.

It should be better to look at sigaction(2) for portability and consistency's sake. Would you be interested in such patch ?

Comment (pickfire) [reply spt.c:84]: Yeah, I am all ears. Feel free to send in a patch.

**Pull request closed without merging**